src/tokenizer/ngram_tokenizer.rs

```diff
@@ -0,0 +1,173 @@
+use super::{Token, TokenStream, Tokenizer};
+
+/// Tokenize the text by splitting words into n-grams of the given size(s)
+///
```

Comment:
Please rename `edges_only` to something that `prefix_only` and document it in the doc.

Also, explicitely state that the position emitted will be the starting offset of the  ngram as this is quite counterintuitive.

Reply:
Ok, should it be something else? I can see now how the other's are working with their `position` fields, and this one is deviating. 🤔

Reply:
I think that's ok for a tokenizer. That's also Lucene behavior. (not for the TokenFilter but for thisf Tokenizer).

I also added #291 too. It's about adding a length to tokens.